src/crypto/index.js

```diff
@@ -266,7 +270,10 @@ function _maybeUploadOneTimeKeys(crypto) {
         now - crypto._lastOneTimeKeyCheck < uploadPeriod
        ) {
         // we've done a key upload recently.
-        return;
+        if (currentKeyCount === undefined) {
+            // only return when currentCount (from /sync-response) is not set
```

Comment:
why? We should also back off when the count is returned from the sync response, otherwise the following could happen:

* say we start with zero keys
* we start uploading the keys
* the server starts sending us a sync response (with key_count == 0)
* we finish uploading the keys
* we receive the sync response
* key_count == 0, so we upload lots more keys!